api/src/main/java/org/openmrs/module/appframework/service/AppFrameworkServiceImpl.java

```diff
@@ -356,6 +369,62 @@ public boolean checkRequireExpression(Extension candidate, AppContextModel conte
         }
     }
 
+    protected String getPatientUuid(AppContextModel contextModel) {
+    	Bindings bindings = new SimpleBindings();
+	    for (Map.Entry<String, Object> e : contextModel.entrySet()) {
+             bindings.put(e.getKey(), e.getValue());
+	    }
+        javascriptEngine.setBindings(bindings, ScriptContext.ENGINE_SCOPE);
+    	try {
+			String uuid = (String) javascriptEngine.eval("patient.uuid;");
+			if (StringUtils.isBlank(uuid)) {
+	    		throw new APIException("Patient uuid cannot be empty");
+			}
+			return uuid;
+		} catch (ScriptException e) {
+    		    	throw new APIException("Failed to get patient uuid", e);
+		}
+    }
+    
+    protected boolean checkRequiredState(Extension candidate, AppContextModel contextModel) {
+    	String requiredState = candidate.getRequiredState();
+    	if (StringUtils.isBlank(requiredState)) {
+    		return true;
+    	}
+    	Concept stateConcept = AppFrameworkUtil.getConcept(requiredState);
+    	if (stateConcept == null) {
+    		throw new APIException("Could not find Concept with id or mapping :" + requiredState);
+    	}
+    	String patientUuid = getPatientUuid(contextModel);
+ 
+    	Patient patient = Context.getPatientService().getPatientByUuid(patientUuid);
```

Comment:
You're un a bean here, you should autowire other beans rather than relying on `Context` getters.

Reply:
You mean something like :

```
@Autowired
PatientService patientService;

public void usePatientService() {
       Patient patient = patientService.getPatient(1);
}

```

??